ssl/statem/extensions_clnt.c

```diff
@@ -524,12 +528,55 @@ int tls_construct_ctos_psk_kex_modes(SSL *s, WPACKET *pkt, X509 *x,
     return 1;
 }
 
-int tls_construct_ctos_key_share(SSL *s, WPACKET *pkt, X509 *x, size_t chainidx,
-                                 int *al)
+static int add_key_share(SSL *s, WPACKET *pkt, unsigned int curve_id)
+{
+    unsigned char *encodedPoint = NULL;
+    EVP_PKEY *key_share_key = NULL;
```

Comment:
Yeah, I know. I've been bitten enough times by travis complaining about an uninit variable (whether justly or not) that I tend to init anyway most often.